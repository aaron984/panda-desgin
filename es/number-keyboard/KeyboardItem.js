import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React from 'react';
import classNames from 'classnames';
import TouchFeedback from 'react-tap-feedback';

var KeyboardItem =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(KeyboardItem, _React$PureComponent);

  function KeyboardItem() {
    _classCallCheck(this, KeyboardItem);

    return _possibleConstructorReturn(this, _getPrototypeOf(KeyboardItem).apply(this, arguments));
  }

  _createClass(KeyboardItem, [{
    key: "render",
    value: function render() {
      var _classNames;

      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          _onClick = _this$props.onClick,
          className = _this$props.className,
          children = _this$props.children,
          type = _this$props.type,
          action = _this$props.action;
      var value = children;

      if (action) {
        value = action;
      }

      var cls = classNames("".concat(prefixCls, "-item"), (_classNames = {
        'panda-hairline': true
      }, _defineProperty(_classNames, "".concat(prefixCls, "-item--gray"), type === 'gray'), _defineProperty(_classNames, "".concat(prefixCls, "-item--middle"), type === 'middle'), _classNames), className);
      return React.createElement(TouchFeedback, {
        activeClassName: "".concat(prefixCls, "-item--active")
      }, React.createElement("span", {
        onClick: function onClick(e) {
          _onClick(value, e);
        },
        className: cls,
        role: "button",
        "aria-label": children
      }, children));
    }
  }]);

  return KeyboardItem;
}(React.PureComponent);

_defineProperty(KeyboardItem, "defaultProps", {
  onClick: function onClick() {},
  disabled: false
});

export { KeyboardItem as default };