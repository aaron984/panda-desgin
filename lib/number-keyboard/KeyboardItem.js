"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactTapFeedback = _interopRequireDefault(require("react-tap-feedback"));

var KeyboardItem =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2["default"])(KeyboardItem, _React$PureComponent);

  function KeyboardItem() {
    (0, _classCallCheck2["default"])(this, KeyboardItem);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(KeyboardItem).apply(this, arguments));
  }

  (0, _createClass2["default"])(KeyboardItem, [{
    key: "render",
    value: function render() {
      var _classNames;

      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          _onClick = _this$props.onClick,
          className = _this$props.className,
          children = _this$props.children,
          type = _this$props.type,
          action = _this$props.action;
      var value = children;

      if (action) {
        value = action;
      }

      var cls = (0, _classnames["default"])("".concat(prefixCls, "-item"), (_classNames = {
        'panda-hairline': true
      }, (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-item--gray"), type === 'gray'), (0, _defineProperty2["default"])(_classNames, "".concat(prefixCls, "-item--middle"), type === 'middle'), _classNames), className);
      return _react["default"].createElement(_reactTapFeedback["default"], {
        activeClassName: "".concat(prefixCls, "-item--active")
      }, _react["default"].createElement("span", {
        onClick: function onClick(e) {
          _onClick(value, e);
        },
        className: cls,
        role: "button",
        "aria-label": children
      }, children));
    }
  }]);
  return KeyboardItem;
}(_react["default"].PureComponent);

exports["default"] = KeyboardItem;
(0, _defineProperty2["default"])(KeyboardItem, "defaultProps", {
  onClick: function onClick() {},
  disabled: false
});