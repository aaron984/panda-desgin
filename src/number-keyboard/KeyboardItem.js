import React from 'react';
import classNames from 'classnames';
import TouchFeedback from 'react-tap-feedback';

export default class KeyboardItem extends React.PureComponent {
  static defaultProps = {
    onClick: () => {},
    disabled: false,
  };

  render() {
    const { prefixCls, onClick, className, children,
      type, action } = this.props;

    let value = children;
    if (action) {
      value = action;
    }
    const cls = classNames(`${prefixCls}-item`, {
      'panda-hairline': true,
      [`${prefixCls}-item--gray`]: type === 'gray',
      [`${prefixCls}-item--middle`]: type === 'middle',
    }, className);

    return (
      <TouchFeedback activeClassName={`${prefixCls}-item--active`}>
        <span
          onClick={(e) => { onClick(value, e); }}
          className={cls}
          role="button"
          aria-label={children}
        >
          {children}
        </span>
      </TouchFeedback>
    );
  }
}
