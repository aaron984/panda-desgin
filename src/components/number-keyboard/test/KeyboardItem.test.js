import React from 'react';
import { render, mount } from 'enzyme';
import KeyboardItem from '../KeyboardItem';

describe('NumberKeyboard and CellGroup', () => {
  it('NumberKeyboard renders correctly', () => {
    const wrapper = render(<KeyboardItem />);
    expect(wrapper).toMatchSnapshot();
  });

  it('NumberKeyboard click event should renders correctly', () => {
    const onClick = jest.fn();
    const wrapper = mount(<KeyboardItem onClick={onClick} />);

    wrapper.find('.panda-number-keyboard-item').simulate('click');
    expect(onClick).toBeCalled();
  });
});
