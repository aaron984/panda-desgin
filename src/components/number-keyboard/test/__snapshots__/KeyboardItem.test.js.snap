// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberKeyboard and CellGroup NumberKeyboard renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "undefined-item panda-hairline",
      "role": "button",
    },
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`NumberKeyboard and CellGroup theme is custom renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "undefined-item panda-hairline",
      "role": "button",
    },
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
